Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -66,6 +67,31 @@ extension MongoCollection {
         return Int(count)
     }
 
+    /**
+     * Counts the number of documents in this collection matching the provided filter.
+     *
+     * - Parameters:
+     *   - filter: a `Document`, the filter that documents must match in order to be counted
+     *   - options: Optional `CountOptions` to use when executing the command
+     *
+     * - Returns: The count of the documents that matched the filter
+     */
+    public func countDocuments(_ filter: Document = [:], options: CountOptions? = nil) throws -> Int {
```

Comment:
`CountDocumentsOptions` instead of `CountOptions`?

Reply:
`typealias` seems best so we can be consistent about the option structs matching the method names. The CRUD spec edits for new count APIs were done hastily, so I think that explains the oversight.

A more general question may be whether it makes sense to use option structs for these things instead of just adding optional arguments to the methods themselves. It may be worth taking a look at the Python and Ruby drivers for inspiration, as both of them use keyword args AFAIK. PHP doesn't have that, so we take all options in a dictionary arg (comparable to these option structs I suppose).

Reply:
> I would be open to adding some helper methods that take in the params directly in cases where there are only a few, but I'm not sure where we would draw the line on that.

These are good points. I'm cool with sticking to option structs and not deviating so we can avoid having to decide where a line is. Let's just keep everything consistent.

Folks can always create their options separately to avoid the first example with a super long line of code.